tools/ds-identify

```diff
@@ -450,7 +449,7 @@ detect_virt() {
 read_virt() {
     cached "$DI_VIRT" && return 0
     detect_virt
-    DI_VIRT=${_RET}
+    DI_VIRT="$(echo "${_RET}" | tr '[:upper:]' '[:lower:]')"
```

Comment:
this is really bad.  We shouldn't do things "just in case".   That doesn't actually make things better, but  makes them worse for a couple reasons:
 * systemd documents the outputs of 'systemd-detect-virt', and they are all  lower case. Accepting upper case is just a waste.
  * it is dramatically slower than it was before (2 additional forks - admittedly forks are fast on linux, but the are much slower than string comparisons).
   * The general policy of "just in case" is silly.  Should we also accept misspellings? in case something spells it 'vmwhere' ?

Reply:
Hi @smoser,

I concur. I thought this was removed prior to merge (see my other comments about reducing this PR to just the bits in `dscheck_VMware`. To be clear, this is not for misspelling, but your point is well received. Even if not systemd (FreeBSD), the mapping table ensures lower case as well.

I propose we revert this line to the former value.

Reply:
Hi @smoser,

Please see https://github.com/canonical/cloud-init/pull/978.

Reply:
thanks.

Reply:
Point taken. Thanks.